pkg/corerp/api/v20231001preview/environment_conversion_test.go

```diff
@@ -545,3 +574,159 @@ func getTestKubernetesEmptyMetadataExtensions(t *testing.T) []datamodel.Extensio
 
 	return extensions
 }
+
+func Test_toRecipeConfigTerraformProvidersDatamodel(t *testing.T) {
+	tests := []struct {
+		name   string
+		config *RecipeConfigProperties
+		want   map[string][]datamodel.ProviderConfigProperties
+	}{
+		{
+			name:   "empty",
+			config: &RecipeConfigProperties{},
+			want:   nil,
+		},
+		{
+			name: "single provider",
```

Comment:
did you mean to add a test case for multiple providers as well? Otherwise I'm not sure what single implies here